content/docs/04.5-devx/05-app-profile/15-db-services/4-connectivity.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Database Connectivity"
+metaTitle: "Palette Dev Engine Database Connectivity"
+metaDescription: "Palette Dev Engine Database Connectivity"
+hideToC: false
+fullWidth: false
+---
+
+import Tabs from 'shared/components/ui/Tabs';
+import WarningBox from 'shared/components/WarningBox';
+import InfoBox from 'shared/components/InfoBox';
+
+
+# Service Connectivity
+
+Connectivity is the process of establishing communication between the database service and the app services. Connectivity is required to send commands and receive answers as result sets from the database. Palette database services establish the connectivity by distributing underlying driver or provider with connection strings, defined as output variables corresponding to individual database services.
+
+The primary considerations in this regard are:
+
+* Each database service comes with service-specific output variables. These output variables in the form of macros are passed to lower-layer app services to establish connectivity.
+
+
+* While adding services to the app profile, consider upward compatibility, where the DB service needs to be at the bottom-most layer (the first layer to be added to the app profile). This is because the variables used in services follow a usage hierarchy where variable values from the first services you add, which become the first layer in the App Profile stack, can pass to services you add next, which appear higher in the stack. However, variable values cannot pass from the top service layers downwards.
```

Comment:
```suggestion
* When adding services to the app profile, consider the order of service layers. Services must be added after the database service layer for other services that require connectivity with the database. In other words, the database service should be at the app profile's bottom-most layer. 

The order of the service layers is important because the output variables used in services follow a usage hierarchy. The output variables for a service are only available if the service comes after the service that exposes the output variable. Output Variables from the first services you add, which become the first layer in the app profile stack, can be consumed by other services after it. However, output variables cannot be passed downwards from the top service layers.
```
I took a stab at simplifying the information. This is a concept we want to make sure everyone understands.